acme_diags/driver/qbo_driver.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import print_function
+
+import numpy as np
+import scipy.fftpack
+import cdutil
+from acme_diags.derivations import acme, default_regions
+from acme_diags.driver import utils
+from acme_diags.plot.cartopy.qbo_plot import plot
+
+
+def process_u_for_time_height(data_region):
+    # Average over longitude (i.e., each latitude's average in data_region)
+    data_lon_average = cdutil.averager(data_region, axis='x')
+    # Average over latitude (i.e., average for entire data_region)
+    data_lon_lat_average = cdutil.averager(data_lon_average, axis='y')
+    # Get data by vertical level
+    level_data = data_lon_lat_average.getAxis(1)
+    return data_lon_lat_average, level_data
+
+
+def deseason(xraw):
+    # Calculates the deseasonalized data
+    months_per_year = 12
+    # Create array to hold climatological values and deseasonalized data
+    # Create months_per_year x 1 array of zeros
+    xclim = np.zeros((months_per_year, 1))
+    # Create array with same shape as xraw
+    x_deseasoned = np.zeros(xraw.shape)
+    # Iterate through all 12 months.
+    for month in np.arange(months_per_year):
+        # `xraw[month::12]` will return the data for this month every year (12 months)
+        # (i.e., from month until the end of xraw, get every 12th month)
+        # Get the mean of this month, using data from every year, ignoring NaNs
+        xclim[month] = np.nanmean(xraw[month::months_per_year])
+    num_years = int(np.floor(len(x_deseasoned) / months_per_year))
+    # Iterate through all years in x_deseasoned (same number as in xraw)
+    for year in np.arange(num_years):
+        year_index = year * months_per_year
+        # Iterate through all months of the year
+        for month in np.arange(months_per_year):
+            month_index = year_index + month
+            # Subtract the month's mean over num_years from xraw's data for this month in this year
+            # i.e., get the difference between this month's value and it's "usual" value
+            x_deseasoned[month_index] = xraw[month_index] - xclim[month]
+    return x_deseasoned
+
+
+def get_20to40month_fft_amplitude(qboN, levelN):
+    # Calculates the amplitude of wind variations in the 20 - 40 month period
+    pspecN = np.zeros(levelN.shape)
+    ampN = np.zeros(levelN.shape)
+
+    for ilev in np.arange(len(levelN)):
+        # `qboN[:, ilev]` returns the entire 0th dimension for ilev in the 1st dimension of the array.
+        y_input = deseason(np.squeeze(qboN[:, ilev]))
+        Y = scipy.fftpack.fft(y_input)
+        n = len(Y)
+        # TODO: f is frequency?
```

Comment:
If so, rename it to `frequency`

Reply:
Yes, f is frequency. Do you want me to rename it here?

Reply:
Thanks! No worries, I'll rename it.